### PR TITLE
cc: Use parameter expansion instead of basename

### PR DESCRIPTION
**Pull request by michaelkuhn**
While debugging #24508, I noticed that we call `basename` in `cc`. The same can be achieved by using Bash's parameter expansion, saving one external process per call.

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
Reference with an explanation for convenience of people reading this PR: https://unix.stackexchange.com/a/171786/21943

**Comment by alalazo** [comment]
I'll be waiting for a second approval before merging this, but LGTM

**Review by becker33 — DISMISSED** [review]

**Comment by adamjstewart** [comment]
Does anyone have performance metrics before and after for comparison?

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by scottwittenburg** [comment]
The most recent gitlab pipeline triggered by @alalazo (via spackbot) failed, as it seems the last time the PR was pushed was several months ago, when we still only had a single stack getting tested on PRs.  So this PR needs to be rebased and force-pushed to test the rest of the stacks.  Also, the errors encountered in the job traces are things the current gitlab ci code base would try to ignore.

Caveat: right now, we've had a job fail in a develop pipeline which will result in other pipelines failing fast when they encounter the same full hash.  That will possibly get resolved automatically when the failed job gets retried and succeeds.  So if you rebase this now, and pipeline generation fails, that might be expected, and in a little while asking spackbot to retrigger the pipeline should result in success.

**Comment by alalazo** [comment]
@michaelkuhn Can you rebase this one? It has been stuck on auto-merge for ages I just noticed today :laughing:

**Comment by michaelkuhn** [comment]
Sure thing! I also totally forgot about this one. :smile:

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by alalazo** [comment]
@michaelkuhn Can you do one rebase more when you have time? Apologies for the pain

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by alalazo** [comment]
Ahahah, it only took 3 months for auto-merge to be triggered!

**Pull request merged**
